src/nvim/testdir/test_ruby.vim

```diff
@@ -0,0 +1,59 @@
+" Tests for ruby interface
+
+if !has('ruby')
+  finish
+end
+
+func Test_ruby_change_buffer()
+  call setline(line('$'), ['1 line 1'])
+  ruby Vim.command("normal /^1\n")
+  ruby $curbuf.line = "1 changed line 1"
+  call assert_equal('1 changed line 1', getline('$'))
+endfunc
+
+func Test_ruby_evaluate_list()
+  call setline(line('$'), ['2 line 2'])
+  ruby Vim.command("normal /^2\n")
+  let l = ["abc", "def"]
+  ruby << EOF
+  curline = $curbuf.line_number
+  l = Vim.evaluate("l");
+  $curbuf.append(curline, l.join("\n"))
+EOF
+  normal j
+  .rubydo $_ = $_.gsub(/\n/, '/')
+  call assert_equal('abc/def', getline('$'))
+endfunc
+
+func Test_ruby_evaluate_dict()
+  let d = {'a': 'foo', 'b': 123}
+  redir => l:out
+  ruby d = Vim.evaluate("d"); print d
+  redir END
+  call assert_equal(['{"a"=>"foo", "b"=>123}'], split(l:out, "\n"))
+endfunc
+
+func Test_rubydo()
+  " Check deleting lines does not trigger ml_get error.
+  new
+  call setline(1, ['one', 'two', 'three'])
+  rubydo Vim.command("%d_")
```

Comment:
```
Caught exception in Test_rubydo(): Vim(return):RuntimeError: Index out of bounds @ function RunTheTest[35]..Test_rubydo[4]..provider#ruby#Call, line 17
```

Reply:
I'm not sure what's supposed to happen here... deleting lines is documented as not supported inside `rubydo`:

> Setting $_ will change the text, but note that it is not possible to add or delete lines using this command.

I will look into making this not raise, but I'm not sure what the actual behavior should be. Is there another way you can reproduce this behavior?

Reply:
I don't know how else to reproduce this behavior. Can you check if this test changes in https://github.com/vim/vim/commit/edd6aacb010ad2402fc98f19e7f6b7c29ba2656f#diff-2fab172276514dfe2a1e9299e003e940?
I can't add this patch yet.